core/primitives/src/shard_layout.rs

```diff
@@ -61,10 +61,12 @@ impl ShardLayout {
         Self::v0(1, 0)
     }
```

Comment:
unrelated, but that `default` feels weird. Seems it wants to be `impl Default for ShardLayout` or maybe renamed to `single_shad` (intuitively, it seems that theer isnt' good default shard layout)?